### ISSUE TITLE
[Upgrade] Routes module errors

### ISSUE THREAD

**Issue by Kaik** [zikula/core#2171]
Hi, today I have tried upgrade zikula and it failed with errors regarding ZikulaRoutesModule
* LogUtil missing use statement https://github.com/zikula/core/blob/1.4/src/system/ZikulaRoutesModule/Api/Base/NotificationApi.php#L87
* all $this->request call to non object https://github.com/zikula/core/blob/1.4/src/system/ZikulaRoutesModule/Api/Base/NotificationApi.php#L92

**Comment by craigh** [comment]
point 1 is fixed in the generator here: https://github.com/Guite/MostGenerator/commit/12e10c38accafb6d623ba8d9ed2a6b0317f3c742

and also in a gigantic pending PR that I am working on.

I'm not understanding point 2 - session is defined in \Zikula_AbstractBase::__construct

**Comment by craigh** [comment]
ah - I just noticed your `[Upgrade]` note.

the entire upgrade is being rewritten. come back to this in a few days :wink:

**Comment by craigh** [comment]
@Guite I think point two on this ticket is a generator issue. the request may not be valid on upgrade. you may want to check `!empty()` or something.

**Comment by Guite** [comment]
We could maybe do something like
```php
if (is_object($this->request) && is_object($this->request->getSession())) {
    $this->request->getSession()->del($this->name . 'AdditionalNotificationRemarks');
} else {
    SessionUtil::del($this->name . 'AdditionalNotificationRemarks');
}
```

I won't add this to the generator though, because this is a special case (generated modules are not called on a core update usually).

**Comment by cmfcmf** [comment]
You could still add it to the generator if the module is a system module.

**Comment by craigh** [comment]
his report was with the old upgrader, so unknown if it still applies. but still may be a worthy addition.

**Comment by Guite** [comment]
@cmfcmf why does the Routes module has another workflow than _NONE_? Otherwise there wouldn't be a notification api generated and used at all.

**Comment by cmfcmf** [comment]
Well, I thought it might be cool if moderators / users could propose routes. I thought "Hey I'm using MOST, why not add a Workflow?".

**Comment by craigh** [comment]
:-1: :boat:

**Comment by craigh** [comment]
@guite hard to regenerate the module without the workflow? Or can you solve this easily another way?

**Comment by Guite** [comment]
Yes, I can do this in the next days.

**Comment by craigh** [comment]
@Guite  - after pulling this change, I tried to install Dizkus and got this error which seems related here:

> [1] Exception: Action: approve not available in this State: initial
    at n/a
        in /Applications/MAMP/htdocs/core.git/src/lib/legacy/Zikula/Workflow.php line 186

>    at Zikula_Workflow->executeAction('approve', object(RouteEntity), 'initial')
        in /Applications/MAMP/htdocs/core.git/src/lib/legacy/Zikula/Workflow/Util.php line 176

>    at Zikula_Workflow_Util::executeAction('none', object(RouteEntity), 'approve', 'route', 'ZikulaRoutesModule', 'id')
        in /Applications/MAMP/htdocs/core.git/src/system/ZikulaRoutesModule/Util/Base/WorkflowUtil.php line 192

>    at Zikula\RoutesModule\Util\Base\WorkflowUtil->executeAction(object(RouteEntity), 'approve')
        in /Applications/MAMP/htdocs/core.git/src/system/ZikulaRoutesModule/Entity/Repository/Route.php line 136

>    at Zikula\RoutesModule\Entity\Repository\Route->addRouteCollection(object(ZikulaDizkusModule), object(RouteCollection))
        in /Applications/MAMP/htdocs/core.git/src/system/ZikulaRoutesModule/Listener/InstallerListener.php line 210

>    at Zikula\RoutesModule\Listener\InstallerListener->addRoutesToCache(object(ZikulaDizkusModule))
        in /Applications/MAMP/htdocs/core.git/src/system/ZikulaRoutesModule/Listener/InstallerListener.php line 80

>    at Zikula\RoutesModule\Listener\InstallerListener->modulePostInstalled(object(ModuleStateEvent), 'module.postinstall', object(Zikula_EventManager))
        in  line 

    at call_user_func(array(object(InstallerListener), 'modulePostInstalled'), object(ModuleStateEvent), 'module.postinstall', object(Zikula_EventManager))
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/Debug/WrappedListener.php line 59

>    at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(ModuleStateEvent), 'module.postinstall', object(Zikula_EventManager))
        in  line 

>    at call_user_func(object(WrappedListener), object(ModuleStateEvent), 'module.postinstall', object(Zikula_EventManager))
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php line 164

>    at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(object(WrappedListener)), 'module.postinstall', object(ModuleStateEvent))
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php line 53

>    at Symfony\Component\EventDispatcher\EventDispatcher->dispatch('module.postinstall', object(ModuleStateEvent))
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/ContainerAwareEventDispatcher.php line 167

>    at Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher->dispatch('module.postinstall', object(ModuleStateEvent), null)
        in /Applications/MAMP/htdocs/core.git/src/lib/legacy/Zikula/EventManager.php line 99

>    at Zikula_EventManager->dispatch('module.postinstall', object(ModuleStateEvent))
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcher.php line 112

>    at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch('module.postinstall', object(ModuleStateEvent))
        in /Applications/MAMP/htdocs/core.git/src/system/Zikula/Module/ExtensionsModule/Controller/AdminController.php line 219

>    at Zikula\Module\ExtensionsModule\Controller\AdminController->viewAction(object(Zikula_Request_Http))
        in  line 

>    at call_user_func_array(array(object(AdminController), 'viewAction'), array(object(Zikula_Request_Http)))
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php line 145

>    at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Zikula_Request_Http), '1')
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php line 66

>    at Symfony\Component\HttpKernel\HttpKernel->handle(object(Zikula_Request_Http), '1', true)
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/ContainerAwareHttpKernel.php line 64

>    at Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle(object(Zikula_Request_Http), '1', true)
        in /Applications/MAMP/htdocs/core.git/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php line 186

>    at Symfony\Component\HttpKernel\Kernel->handle(object(Zikula_Request_Http))
        in /Applications/MAMP/htdocs/core.git/src/index.php line 25

**Comment by craigh** [comment]
the result is that Dizkus appears to be installed, but the routes are not loaded

**Comment by craigh** [comment]
if I try to reload the routes, I get the same error.

**Comment by Guite** [comment]
Fixed in 3e633724cd23cf4d6cd4e76c0e086bf57fa28c9b

**Comment by craigh** [comment]
confirmed. thank you.
